### PR TITLE
Expand boilerplate and make customizable for different kinds of plugins.

### PR DESCRIPTION
**Pull request by rtibbles**
Expands the boilerplate code and makes most options more customizable.

Also cuts out unnecessary code depending on the purpose of the plugin.

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -1,4 +1,15 @@
 {
-  "plugin_name": "example_plugin",
-  "plugin_class_name": "ExamplePlugin"
+  "readable_plugin_name": "Example Plugin",
+  "description": "",
+  "repository_url": "",
+  "author": "Learning Equality",
+  "email": "info@learningequality.org",
+  "license": ["MIT", "GNU GPL v3.0", "Apache Software License 2.0", "The Unlicense"],
```
Comment (rtibbles) [inline cookiecutter.json]: Comment from @jamalex that we probably want to remove GPL from here as it makes reusing others' plugins problematic.

Can also collapse to a hard default of MIT that people have to modify if they want to change the license.

File: README.md [hunk 2]
```diff
@@ -7,21 +7,33 @@ Minimal [Cookiecutter](https://github.com/audreyr/cookiecutter) template for gen
...
   `$ cookiecutter https://github.com/learningequality/cookiecutter-kolibri-plugin`
 
-
-* Select a plugin name. 
+* Select a readable plugin name. 
   
-  Must be `lowercase_separated_by_underscores` (ie. `vector_video_player`).
+  e.g. `Vector Video Player`.
+
+* You will then be prompted to enter a description, a repository url, and author details..
```
Comment (christianmemije) [inline README.md]: Extra period.

File: README.md [hunk 3]
```diff
@@ -7,21 +7,33 @@ Minimal [Cookiecutter](https://github.com/audreyr/cookiecutter) template for gen
...
-* Select a plugin Class name. 
+* If you would like to change the default plugin Class name for the Javascript class, you can modify it. 
   
   Must be `UpperCamelCase.` (ie. `VectorVideoPlayer`).
 
+* Select if this is a frontend plugin (i.e. it will provide a Javascript module for frontend code)
+
+* Select if this plugin will have its own page within the Django app - if you are creating an extension of existing functionality in Kolibri, this will not be the case - such as a content renderer plugin.
+
+* Select if this plugin is a content renderer plugin, and what content kind and file extension it handles. If you are not creating a content renderer plugin, you may safely ignore these options.
```
Comment (christianmemije) [inline README.md:36]: There is no option to add a new content type?

File: {{cookiecutter.plugin_name}}/README.rst [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
+2. Open terminal in your Kolibri repo.
+
+3. run the following commands:
+
+    ``pip install -e <LOCAL-PATH-TO-REPO>``
+
+    ``kolibri plugin {{ cookiecutter.plugin_name }} enable``
+
+{% if is_frontend_plugin %}
+4. Then run the commands to install frontend packages in Kolibri, this plugin will have its recursively installed.
```
Comment (christianmemije) [inline {{cookiecutter.plugin_name}}/README.rst]: Needs clarification.

File: {{cookiecutter.plugin_name}}/README.rst [hunk 5]
```diff
@@ -1,19 +1,51 @@
...
+{% endif %}
+
+How to publish to PyPi?
+------------------------------
 
-Run the following commands in Kolibri to install this plugin:
+1. Follow the instructions above to installing the plugin for development.
```
Comment (christianmemije) [inline {{cookiecutter.plugin_name}}/README.rst]: Typo?

File: {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js [hunk 6]
```diff
@@ -1,25 +1,38 @@
+const RootComponent = require('./vue');
```
Comment (christianmemije) [inline {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js]: `1:7   error  'RootComponent' is defined but never used  no-unused-vars`

File: {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js [hunk 7]
```diff
@@ -1,25 +1,38 @@
...
    */
   ready() {
-    this.vm = new Vue({
-      el: 'body',
-      components: {
-        rootvue: require('./vue'),
-      },
+    this.rootvue = new Vue({
+      el: 'rootvue',
+      render: createElement => createElement(RootVue),
```
Comment (christianmemije) [inline {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js:33]: `21:46  error  'RootVue' is not defined                   no-undef`

**Review by christianmemije — CHANGES_REQUESTED** [review]
Some lint errors.

File: README.md (hunk 3 above)
Comment (rtibbles) [reply README.md:36]: No, the content types are validated against currently allowed content types in `le-utils`.

File: {{cookiecutter.plugin_name}}/README.rst (hunk 4 above)
Comment (rtibbles) [reply {{cookiecutter.plugin_name}}/README.rst]: Should I link to the Kolibri docs?

File: {{cookiecutter.plugin_name}}/README.rst (hunk 5 above)
Comment (rtibbles) [reply {{cookiecutter.plugin_name}}/README.rst]: Definitely!

File: {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js (hunk 6 above)
Comment (rtibbles) [reply {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js]: Good catch.

File: {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js (hunk 7 above)
Comment (rtibbles) [reply {{cookiecutter.plugin_name}}/{{cookiecutter.plugin_name}}/assets/src/module.js:33]: Will standardize.

File: {{cookiecutter.plugin_name}}/README.rst (hunk 4 above)
Comment (christianmemije) [reply {{cookiecutter.plugin_name}}/README.rst]: *"will have its [dependencies?] recursively installed."

**Pull request merged**